components/script/script_thread.rs

```diff
@@ -1129,6 +1307,56 @@ impl ScriptThread {
         }
     }
 
+    fn message_to_pipeline(&self, msg: &MixedMessage) -> Option<PipelineId> {
+        match *msg {
+            MixedMessage::FromConstellation(ref inner_msg) => inner_msg.pipeline(),
+            MixedMessage::FromDevtools(ref inner_msg) => inner_msg.pipeline(),
+            MixedMessage::FromScript(_) => None,
+            MixedMessage::FromImageCache((pipeline_id, _)) => Some(pipeline_id),
+            MixedMessage::FromScheduler(ref timer_event) => {
+                let TimerEvent(source, _) = *timer_event;
+                match source {
+                    TimerSource::FromWindow(pipeline_id) => Some(pipeline_id),
+                    _ => None
+                }
+            }
+        }
+    }
+
+    /// Go through documents and see if any don't have a time to interactive yet
+    /// If not, get all tasks for the pipeline that have been running for > 50ms
```

Comment:
The TTI metric is concerned with a 50ms window that contains any tasks that run for >10ms, and I think the logic described here (and implemented) is backwards.

Reply:
Ack, I totally misremembered how TTI is defined. Sorry about that! This description is correct that we care about >50ms tasks in a 10s window.